topcoffea/modules/corrections.py

```diff
@@ -194,6 +195,58 @@ def GetBTagSF(eta, pt, flavor, year=2018, sys=0):
 
   return (SF)
 
+###### Pileup reweighing
+##############################################
+## Get central PU data and MC profiles and calculate reweighting
+## Using the current UL recommendations in:
+##   https://twiki.cern.ch/twiki/bin/viewauth/CMS/PileupJSONFileforData
+##   - 2018: /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions18/13TeV/PileUp/UltraLegacy/
+##   - 2017: /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions17/13TeV/PileUp/UltraLegacy/
+##   - 2016: /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions16/13TeV/PileUp/UltraLegacy/
+##
+## MC histograms from:
+##    https://github.com/CMS-LUMI-POG/PileupTools/
+
+pudirpath = topcoffea_path('data/pileup/')
+
+def GetDataPUname(year='2017', var=0):
+  ''' Returns the name of the file to read pu observed distribution '''
+  if   var== 0: ppxsec = 69200
+  elif var== 1: ppxsec = 72400
+  elif var==-1: ppxsec = 66000
```

Comment:
Two quick questions:

1. Should we hard code these, or maybe just write the formula?
2. These look like 10%. I thought the pp uncertainty was ~5%.

Reply:
Another quick comment on this, I think numbers like these would be good to keep in a centralized location, e.g. the [params.json](https://github.com/TopEFT/topcoffea/blob/PUweight/topcoffea/json/params.json) file. You can easily access the numbers in that file using the [get_cut()](https://github.com/TopEFT/topcoffea/blob/695125a6af1c9447de0d165eceb602dbc07f6feb/topcoffea/modules/GetValuesFromJsons.py#L13) function from `GetValuesFromJsons.py`, though we should probably think about renaming that function at some point (since not all of the numbers inside that file are cuts).

Reply:
1. Those ppxsec values are introduced in the .root files used to get the PU profile. We left the name they had in the official repository and then: PileupHistogram-goldenJSON-13tev-%s-%sub-99bins.root'%((year), str(ppxsec)

2. You are right, pp uncertainty is ~5% and is what we have:
>>> (72400.-69200)/69200.*100
4.624277456647398
&
>>> (66000.-69200)/69200.*100
-4.624277456647398

Reply:
Ok, thanks. So we have to keep the hard coded values for the file names, that's fine. I forgot to think of the uncertainties as shifts, so thanks for pointing out the correct equation!

Reply:
Should we still follow @kmohrman's comment about putting these in the `params.json` file, or would you prefer they stay hard coded here since they go into the filename?

Reply:
Ok, leaving them here is fine with me. @kmohrman do you have any preference, now that we know they're needed to create the filename string?

Reply:
I think I'd probably still prefer that they were stored in the param json, if it's not too much trouble.

Reply:
Ok,  I'll do it. Do you want me to change the function name to, for example, "get_param()"? (instead of "get_cut()")

Reply:
Sure, if you want do do that, I think that'd be great. Though you'd have to make sure to change it everywhere it gets used.

Reply:
Done!

Reply:
Thanks @andrea21z! I'm running the `check_yields` script on my end now, just to make sure everything works from a fresh pull.

Reply:
Thanks @andrea21z. @bryates, I'm was actually going to run the same test. I'll can plan to push the results (just to make sure you and I don't both try to push at the same time), unless you would prefer to.

Reply:
Sounds good, you can take care of it.